src/imitation/data/types.py

```diff
@@ -288,8 +289,20 @@ def map_maybe_dict(fn, maybe_dict):
         return fn(maybe_dict)
 
 
-# TODO: maybe should support DictObs?
-TransitionMapping = Mapping[str, AnyTensor]
+class TransitionMappingNoNextObs(TypedDict):
+    """Dictionary with `obs` and `acts`."""
+
+    obs: Union[Observation, th.Tensor]
+    acts: AnyTensor
+
+
+# inheritance with total=False so these are not required
+class TransitionMapping(TransitionMappingNoNextObs, total=False):
+    """Dictionary with `obs` and `acts`, maybe also `next_obs`, `dones`, `rew`."""
+
+    next_obs: AnyTensor
```

Comment:
```suggestion
    next_obs: Union[Observation, th.Tensor]
```
for consistency with `obs` above?